### ISSUE TITLE
Dlv headless mode produce many zombie process.

### ISSUE THREAD

**Issue by win5do** [go-delve/delve#2689]
# dlv version: 
```
Delve Debugger
Version: 1.7.1
Build: $Id: 3bde2354aafb5a4043fd59838842c4cd4a8b6f0b $
```

# go version:
```
go version go1.15.12 darwin/amd64
```

# OS:
macOS 11.5.2

# Step

main.go:
```go
package main

import "github.com/gin-gonic/gin"

func main() {
	r := gin.Default()
	r.GET("/ping", func(c *gin.Context) {
		c.JSON(200, gin.H{
			"message": "pong",
		})
	})
	r.Run() // listen and serve on 0.0.0.0:8080 (for windows "localhost:8080")
}
```

start dlv:
```sh
dlv debug --headless --listen=:2345 --api-version=2 --accept-multiclient
```

exit without use remote call，ctrl+c not work，so I use kill:
```sh
kill <dlv-pid>
```

after repeat several times, ps output: 
```
  501 66718     1   0  2:27PM ??         0:00.00 (__debug_bin)
  501 67191     1   0  2:30PM ??         0:00.00 (__debug_bin)
  501 67739     1   0  2:35PM ??         0:00.00 (__debug_bin)
```

# Expect

Handle child process exit.

**Comment by aarzilli** [comment]
Should be fixed by 5d158820f69484106bee28f025bb6660c80bd6e3

**Comment by aarzilli** [comment]
Reopen the issue if it isn't.
